### ISSUE TITLE
No probability return with multi:softprob

### ISSUE THREAD

**Issue by uzhao** [dmlc/xgboost#4434]
I tested below code and expect a matrix return with m.predict(x), however it output same result like softmax. Do I need a one-hot encoding input for y here?

m = xgboost.XGBClassifier(objective='multi:softprob', tree_method='gpu_exact')
m.fit(x, y)
Out[2]: 
XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
       colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,
       max_depth=3, min_child_weight=1, missing=None, n_estimators=100,
       n_jobs=1, nthread=None, objective='multi:softprob', random_state=0,
       reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
       silent=True, subsample=1, tree_method='gpu_exact')
m.predict(x).shape
Out[3]: (124744,)
x.shape
Out[4]: (124744, 120)

**Comment by hcho3** [comment]
Try using `m.predict_proba()` instead.

**Comment by uzhao** [comment]
I know predict_proba, but shouldn't softprob return probability?

**Comment by hcho3** [comment]
@uzhao XGBClassifier follows the convention of scikit-learn, where `predict()` always returns label predictions and `predict_proba()` always returns probability predictions
